### PR TITLE
initial sample

### PR DESCRIPTION
**Pull request by derisen**
cc @rwike77

## Purpose

* Initial sample

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ x ] Other... Please describe: new sample
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

### CODE REVIEW

File: 1-WebApp-storage-managed-identity/README.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+                    </tr>
+                </thead>
+                <tbody id="claims-table">
+                </tbody>
+            </table>
+        </div>
+
+        <% if(user.isLoggedIn) { %>
+            <script>
+                fetch('https://derisen-easyauth-nodejs.azurewebsites.net/.auth/me')
```
Comment (derisen) [inline 1-WebApp-storage-managed-identity/README.md]: hardcoded url -make sure to parametrize all instances in a settings file

File: 1-WebApp-storage-managed-identity/views/id.ejs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js"
+            integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo"
+            crossorigin="anonymous"></script>
+        <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js"
+            integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6"
+            crossorigin="anonymous"></script>
+
+        <% if(user.isLoggedIn) { %>
+            <script>
+                fetch('https://derisen-easyauth-nodejs.azurewebsites.net/.auth/me')
```
Comment (derisen) [inline 1-WebApp-storage-managed-identity/views/id.ejs:43]: hardcoded url -make sure to parametrize in a settings file

File: 1-WebApp-storage-managed-identity/views/includes/navbar.ejs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar navbar-expand-lg navbar-dark bg-dark">
+    <a class="navbar-brand" href="/">Microsoft identity platform</a>
+    <div class="btn-group ml-auto dropleft">
+        <% if (user.isLoggedIn) { %>
+            <a class="btn btn-secondary" href="/id" role="button">ID</a>
+            <a class="btn btn-success" href="/.auth/logout?post_logout_redirect_uri=https://derisen-easyauth-nodejs.azurewebsites.net/logout" role="button">Sign-out</a>
```
Comment (derisen) [inline 1-WebApp-storage-managed-identity/views/includes/navbar.ejs:6]: hardcoded url -make sure to parametrize in a settings file

File: 2-WebApp-graphapi-on-behalf/README.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+                    </tr>
+                </thead>
+                <tbody id="claims-table">
+                </tbody>
+            </table>
+        </div>
+
+        <% if(user.isLoggedIn) { %>
+            <script>
+                fetch('https://derisen-easyauth-nodejs.azurewebsites.net/.auth/me')
```
Comment (derisen) [inline 2-WebApp-graphapi-on-behalf/README.md:168]: hardcoded url -make sure to parametrize in a settings file

File: 2-WebApp-graphapi-on-behalf/views/id.ejs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js"
+                integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo"
+                crossorigin="anonymous"></script>
+            <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js"
+                integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6"
+                crossorigin="anonymous"></script>
+
+            <% if(user.isLoggedIn) { %>
+                <script>
+                    fetch('https://derisen-easyauth-nodejs.azurewebsites.net/.auth/me')
```
Comment (derisen) [inline 2-WebApp-graphapi-on-behalf/views/id.ejs:43]: hardcoded url -make sure to parametrize in a settings file

File: 2-WebApp-graphapi-on-behalf/views/includes/navbar.ejs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar navbar-expand-lg navbar-dark bg-dark">
+    <a class="navbar-brand" href="/">Microsoft identity platform</a>
+    <div class="btn-group ml-auto dropleft">
+        <% if (user.isLoggedIn) { %>
+            <a class="btn btn-secondary" href="/id" role="button">ID</a>
+            <a class="btn btn-success" href="/.auth/logout?post_logout_redirect_uri=https://derisen-easyauth-nodejs.azurewebsites.net/logout" role="button">Sign-out</a>
```
Comment (derisen) [inline 2-WebApp-graphapi-on-behalf/views/includes/navbar.ejs:6]: hardcoded url -make sure to parametrize in a settings file

File: 3-WebApp-graphapi-managed-identity/README.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+                    </tr>
+                </thead>
+                <tbody id="claims-table">
+                </tbody>
+            </table>
+        </div>
+
+        <% if(user.isLoggedIn) { %>
+            <script>
+                fetch('https://derisen-easyauth-nodejs.azurewebsites.net/.auth/me')
```
Comment (derisen) [inline 3-WebApp-graphapi-managed-identity/README.md:168]: hardcoded url -make sure to parametrize in a settings file

File: README.md [hunk 8]
```diff
@@ -1,57 +1,99 @@
...
+In this step, you configure authentication for a web app and [call Microsoft Graph as the web app (using managed identities)](https://github.com/Azure-Samples/ms-identity-easyauth-nodejs-storage-graphapi/tree/main/3-WebApp-graphapi-managed-identity)
 
-### Quickstart
-(Add steps to get up and running quickly)
+You learn how to:
+
+- Setup authentication for your web app.
+- Create a system-assigned managed identity on a web app.
+- Add Microsoft Graph API permissions to a managed identity.
+- Call Microsoft Graph from a web app (on behalf of the app) by using managed identities.
```
Comment (kalyankrishna1) [inline README.md:65]: is this section redundant?

File: README.md [hunk 9]
```diff
@@ -1,57 +1,99 @@
...
 
-- OS
-- Library version
-- ...
+If you are a Line of Business (LOB) developer, you'd probably want to only sign-in users in your organization with their work or school accounts. In this step, you learn how to enable the Azure App Service authentication/authorization module in Azure App Service, so you can sign in users and access data by writing minimal or no code in your web app. This step is required and is done inline with the other steps. Read [Add authentication to your web app running on Azure App Service](https://docs.microsoft.com/azure/app-service/scenario-secure-app-authentication-app-service) for more information.
 
-### Installation
+## Securely access Azure storage on behalf of the web application using managed identities​
+
+In this step, you configure authentication for a web app and [access Azure blob storage as the web app (using managed identities)](https://github.com/Azure-Samples/ms-identity-easyauth-nodejs-storage-graphapi/tree/main/1-WebApp-storage-managed-identity).
```
Comment (kalyankrishna1) [inline README.md]: >https://github.com/Azure-Samples/ms-identity-easyauth-nodejs-storage-graphapi/tree/main/1-WebApp-storage-managed-identity [](http://example.com/codeflow?start=131&length=121)

use relative links

File: 1-WebApp-storage-managed-identity/README.md [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+- [Node.js v12](https://nodejs.org) or later
+- An [Azure subscription](https://docs.microsoft.com/azure/guides/developer/azure-developer-guide#understanding-accounts-subscriptions-and-billing) and an [Azure AD tenant](https://docs.microsoft.com/azure/active-directory/develop/quickstart-create-new-tenant) with one or more user accounts in the directory
+
+### Step 1: Clone or download this repository
+
+Clone or download this repository. From your shell or command line:
+
+```console
+git clone https://github.com/Azure-Samples/ms-identity-easyauth-nodejs-storage-graphapi.git
+ms-identity-easyauth-nodejs-storage-graphapi
```
Comment (kalyankrishna1) [inline 1-WebApp-storage-managed-identity/README.md]: what is this for?

Comment (derisen) [reply 1-WebApp-storage-managed-identity/README.md]: oops, should have been `cd`

File: 1-WebApp-storage-managed-identity/README.md [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+
+Wait for a few minutes for your changes on **App Service** to take effect.
+
+### Step 5: Enable managed identity on an app
+
+Read this article to learn how to [enable a managed identity on a web app](https://docs.microsoft.com/azure/app-service/scenario-secure-app-access-storage#enable-managed-identity-on-an-app).
+
+### Step 6: Grant access to the storage account
+
+You need to grant your web app access to the storage account before you can create, read, or delete blobs. In a previous step, you configured the web app running on App Service with a managed identity. Using Azure RBAC, you can give the managed identity access to another resource, just like any security principal. The Storage Blob Data Contributor role gives the web app (represented by the system-assigned managed identity) read, write, and delete access to the blob container and data. Read this article to learn how to [grant access to the storage account](https://docs.microsoft.com/azure/app-service/scenario-secure-app-access-storage#grant-access-to-the-storage-account).
```
Comment (kalyankrishna1) [inline 1-WebApp-storage-managed-identity/README.md:92]: > your web app access [](http://example.com/codeflow?start=17&length=20)

app, user or both ?

**Review by mistermik — CHANGES_REQUESTED** [review]
Change views/home.ejs to views/id.ejs README.md line165

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

File: README.md (hunk 9 above)
Comment (derisen) [reply README.md]: done

File: 1-WebApp-storage-managed-identity/README.md (hunk 11 above)
Comment (derisen) [reply 1-WebApp-storage-managed-identity/README.md:92]: here its app

File: README.md (hunk 8 above)
Comment (derisen) [reply README.md:65]: No this list matches the [backing tutorial's sections](https://docs.microsoft.com/en-us/azure/app-service/scenario-secure-app-overview)

File: 1-WebApp-storage-managed-identity/README.md [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+
+### Step 7: Visit the web app
```
Comment (kalyankrishna1) [inline 1-WebApp-storage-managed-identity/README.md:94]: >## Step 7: Visit the web app [](http://example.com/codeflow?start=1&length=28)

Expnad this to mention the "comments" button

File: 2-WebApp-graphapi-on-behalf/README.md [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+### Step 2: Deploy the web app and configure App Service authentication
+
+This project has one WebApp project. To deploy it to Azure App Service, you'll need to:
+
+- configure a deployment user
+- create an Azure App Service plan
+- create a web app
+- publish the web app to Azure
+
+For information on how to do this from VS Code using the [App Service Extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azureappservice), see the [tutorial](https://docs.microsoft.com/azure/developer/javascript/tutorial/deploy-nodejs-azure-app-service-with-visual-studio-code?tabs=bash).
```
Comment (derisen) [inline 2-WebApp-graphapi-on-behalf/README.md:58]: Add short summary for deployment steps

File: 1-WebApp-storage-managed-identity/README.md [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+
+Open a browser and navigate to the deployed web app (replace *web-app-name* with the name of your web app): `https://web-app-name.azurewebsites.net/`
+
+## About the code
+
+This sample is built using the [@azure/identity](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/README.md) and [@azure-samples/msal-express-wrapper](https://github.com/Azure-Samples/msal-express-wrapper) packages for authentication and authorization.
+
+### Add authentication to your web app
```
Comment (kalyankrishna1) [inline 1-WebApp-storage-managed-identity/README.md:102]: you might need to be more prescriptive here, like what needs adding to a web project to enable easyauth

**Comment by aremo-ms** [comment]
When running ps script for Graph permissions - I had to login before it, otherwise web app is not found. The script should be re-arranged not to ask for login again. Finally, permission was not set for me.

**Comment by aremo-ms** [comment]
Remove step 4 - it's done in instructions of step 3

File: 1-WebApp-storage-managed-identity/README.md (hunk 12 above)
Comment (derisen) [reply 1-WebApp-storage-managed-identity/README.md:94]: done

**Pull request merged**